### PR TITLE
fix #112736 a few benign compiler warnings

### PR DESCRIPTION
**Pull request by ericfont**

### CODE REVIEW

**Comment by ericfont** [comment]
funny, the test fail because of an extra <System> & </System> in parts/part-breath-add.mscx, but I don't think that is due to this PR. I'm also seeing same error in PR 2625.

**Comment by Jojo-Schmitz** [comment]
a rebase is needed

File: omr/omrpage.cpp [hunk 1]
```diff
@@ -547,7 +547,7 @@ namespace Ms {
                 p.sym  = pattern->id();
                 p.prob = val;
             }
-            printf("Pattern found %d %f %d\n", pattern->id(), val, xx);
+            printf("Pattern found %d %f %d\n", (int) pattern->id(), val, xx);
```
Comment (Jojo-Schmitz) [inline omr/omrpage.cpp]: maybe better `int(pattern->id())`, Looks more C++ish

File: thirdparty/ofqf/qoscserver.cpp [hunk 2]
```diff
@@ -19,128 +19,128 @@
...
+                        path.replace( rep, replacements[ rep ] );
+
+                  //qDebug() << " after transformation to OSC-RegExp path is" << path;
+
+                  QRegExp exp( path );
+                  foreach( PathObject* obj, paths ) {
+                        if ( exp.exactMatch( obj->_path ) )
+                              obj->signalData( arguments );
+                        }
+                  }
```
Comment (Jojo-Schmitz) [inline thirdparty/ofqf/qoscserver.cpp]: I don't think we want to change too much on third party stuff, would potentially make it difficult to update from upstream

File: libmscore/barline.cpp [hunk 3]
```diff
@@ -1272,10 +1272,11 @@ QString BarLine::accessibleExtraInfo() const
                   for (const Element* e : nextM->el()) {
                         if (!score()->selectionFilter().canSelect(e))
                               continue;
-                        if (e->isMarker())
+                        if (e->isMarker()) {
                               if (toMarker(e)->markerType() == Marker::Type::FINE)
                                     continue; //added above^
                               rez = QString("%1 %2").arg(rez).arg(e->screenReaderInfo());
+                              }
```
Comment (Jojo-Schmitz) [inline libmscore/barline.cpp]: this changes symantic, although it really it looks like this is what this code was supposed to be

File: thirdparty/ofqf/qoscserver.cpp (hunk 2 above)
Comment (ericfont) [inline thirdparty/ofqf/qoscserver.cpp]: ok.  I did mention that concern in the issue report.  I guess I can just make the minimal change needed for this specific warning.

File: libmscore/barline.cpp (hunk 3 above)
Comment (ericfont) [inline libmscore/barline.cpp]: I mentioned that in the issue report.  As I said I believe this is what the code is supposed to do, but whoever approves this pr needs to make sure.

**Comment by Jojo-Schmitz** [comment]
Travis mtest failure should be fixed with eaa0322

**Comment by ericfont** [comment]
ok, rebased with eaa0322.

File: libmscore/barline.cpp (hunk 3 above)
Comment (MarcSabatella) [inline libmscore/barline.cpp]: I could look at this later.

Comment (MarcSabatella) [inline libmscore/barline.cpp]: Looks like either way works, just differently.  The way it actually was, the screenreader would read *all* elements attached to the next measure, except for Markers of type Fine.  The way it *looks* like it was, it would only read *markers*, again except for ones of type Fine.

Meaning, with a barline selected, if something was attached to the next measure that was *not* a marker, it would be read, and the proposed changed would stop that.  Without knowing the full list of things that could conceivably be attached to measures, it's hard to say whether you'd *want* other things read or not.  Since we don't read anything else attached to *this* measure except jumps and Fine markers, I am inclined to think we don't want things attached to the *next* measure read either.

Bigger picture: I guess the point here is to read "coda" when selecting the barline before the coda.  This really only makes sense if the coda measure directly follows the previus measure, no intervening line break or horizontal frame (because otherwise the barline at the end of the measure before the coda would not be physically attached to the coda).  As it is, it looks like we'll actually read "coda" when selecitng the barline at the end of a line if the next system starts with a coda.  I'm not sure that makes sense.  So we could also consider removing this whole loop.

Bottom line: it's going to work one way or another, so it doesn't matter much.  @andreituicu feel free to weigh in if you can remember your reasoning.

Right now, my inclination is to remove the whole loop.  Maybe replace this section of code with something that really figures out if this barline is at the start or end of the measure and reads accordingly.  But I also want to implement a command to navigate all elements directly, meaning we eventually won't want to read anything else attached to the measure just because a barline is selected.

File: omr/omrpage.cpp [hunk 4]
```diff
@@ -471,7 +471,7 @@ OmrPattern OmrPage::searchPattern(const std::vector<Pattern*>& pl, int y, int x1
                   p.sym = pattern->id();
                   p.prob = val;
                   }
-            printf("Pattern found %d %f %d\n", pattern->id(), val, xx);
+            printf("Pattern found %d %f %d\n", int(pattern->id()), val, xx);
```
Comment (Jojo-Schmitz) [inline omr/omrpage.cpp]: this is fixed in 1ed009e now (and in exactly the same way, so no conflict)

File: libmscore/barline.cpp (hunk 3 above)
Comment (andreituicu) [inline libmscore/barline.cpp]: From what I remember, I wanted the screen reader to say the Markers and Jumps as extraInfo when a barline was selected. I think that the proposed change fixes a bug of mine.

@MarcSabatella: I wouldn't remove the loop. In my opinion, even if at some point there would be a command for navigating all elements directly, my idea was to have the following workflow, based on how I've seen people read scores in general (to be clear: not visually impaired musicians):
- The user uses the existing command to familiarise with the main elements of the score (Clefs, Time Signatures, Notes, Rests, Barlines, etc.) and he can skip the extra info.
- After he has a good idea about the melody and rhythm, he can traverse the score again and listen to the extra info, meaning elements that are attached.
- And with a new command the user can have the possibility to select each individual element, but I was thinking that is mainly for editing.

In my opinion it is more natural to have the screen reader say all the elements that are attached to the selected one as extra info, instead of having to select each element and figure out to what they are attached to based on bar and beat. I would be best to have some feedback from users on this.

**Comment by MarcSabatella** [comment]
@andreituicu I think you may have misunderstood why I wanted to remove the loop.  I agree that for now at least we want to read markings attached to selected elements.  The issue here here, the marking in question is *not* actually attached to the selected barline.  It's attached to the *next measure*.  So selecting a barline at the end of one measure will cause markings attachjed to the *next* measure to be read.  This might make sense for codas if the two measures are physically adjacent - so they share the barline - but it does not make sense if the measures are separated by a frame or line break.  And in the case of a coda - which is, I assume, the main case we are concerned with here - there would almost always in fact be either a frame or a line break.

Unless there is some other reason I am missing why we are actually reading elements attached to the next measure - again, I am not sure what the full list of such elements might even be.
